src/charm.py

```diff
@@ -88,12 +98,18 @@ def _process_env_file(self, event):
         Args:
             event: The event triggered when the relation changed.
         """
-        try:
+        if not self._state.is_ready():
+            event.defer()
+            return
+
+        if self.unit.is_leader():
             self._state.env = None
```

Comment:
What happens if you get a `ModelError` but you already set `env` to `None`?

Reply:
```ModelError``` happens when the resource has not been attached yet, so this error should not happen if a user has attached a new resource.